lib/irb/cmd/debug.rb

```diff
@@ -82,14 +82,22 @@ def DEBUGGER__.capture_frames(*args)
       # it's a bundled gem. This method tries to activate and load that.
       def load_bundled_debug_gem
         # Discover latest debug.gem under GEM_PATH
-        debug_gem = Gem.paths.path.map { |path| Dir.glob("#{path}/gems/debug-*") }.flatten.select do |path|
-          File.basename(path).match?(/\Adebug-\d+\.\d+\.\d+\z/)
+        debug_gem = Gem.paths.path.flat_map { |path| Dir.glob("#{path}/gems/debug-*") }.select do |path|
+          File.basename(path).match?(/\Adebug-\d+\.\d+\.\d+(\w+)?\z/)
         end.sort_by do |path|
           Gem::Version.new(File.basename(path).delete_prefix('debug-'))
         end.last
         return false unless debug_gem
 
+        # Discover debug/debug.so under extensions for Ruby 3.2+
+        debug_so = Gem.paths.path.flat_map do |path|
```

Comment:
Do we need to avoid triggering this path for Ruby 3.2-?

Reply:
No. It seems like the original path, and it's intentional that it uses that path for older versions as well.